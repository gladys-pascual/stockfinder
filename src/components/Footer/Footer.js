import "./Footer.scss";

const Footer = () => {
  return (
    <footer className="footer-section">
      <a
        href="https://finnhub.io/docs/api"
        target="_blank"
        rel="noopener noreferrer"
        className="finnhub"
      >
        <div
          className="finnhub-logo"
          href="https://finnhub.io/docs/api"
          target="_blank"
          rel="noopener noreferrer"
        >
          <img
            className="finnhub-logo-img"
            src="https://static.finnhub.io/img/finnhub_2020-05-09_20_51/logo/logo-gradient-thumbnail-trans.png"
            alt="Finnhub logo"
          />
        </div>
        <p className="finnhub-text">This website uses the Finnhub API.</p>
      </a>

      <ul className="footer-socials">
        <li>
          <a
            href="https://github.com/gladys-pascual/"
            target="_blank"
            rel="noopener noreferrer"
          >
            <i className="fab fa-github">
              <span className="sr-only">GitHub icon</span>
            </i>
          </a>
        </li>
        <li>
          <a
            href="https://dev.to/gladyspascual"
            target="_blank"
            rel="noopener noreferrer"
          >
            <i className="fab fa-dev">
              <span className="sr-only">Dev icon</span>
            </i>
          </a>
        </li>
        <li>
          <a
            href="https://www.linkedin.com/in/gladyspascual/"
            target="_blank"
            rel="noopener noreferrer"
          >
            <i className="fab fa-linkedin">
              <span className="sr-only">LinkedIn icon</span>
            </i>
          </a>
        </li>
      </ul>
    </footer>
  );
};

export default Footer;
