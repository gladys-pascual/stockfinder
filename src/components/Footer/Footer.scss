@import "../../assets/variables.scss";

.footer-section {
  background-color: $theme-color;
  padding: 8px 0px;
  margin-top: 80px;
  display: flex;
  align-items: center;
  color: $text-color-hover;
  flex-direction: column;

  .footer-socials {
    padding: 0px 100px;
    display: flex;
    margin-bottom: 0px;
    @media screen and (max-width: 850px) {
      padding: 0px 12px;
    }
    li {
      list-style-type: none;
      padding: 5px;
      display: inline-block;
      a {
        display: inline-block;
        height: 44px;
        width: 44px;
        background-color: #ffffff;
        color: black;
        border-radius: 50%;
        margin: 0 2px;
        text-align: center;
        line-height: 42px;
        font-size: 28px;
        transition: all 0.5s ease;
      }
      a:hover {
        box-shadow: 0 0 10px #000;
      }
      .finnhub {
        display: flex;
        justify-content: center;
        width: 160px;
        border-radius: 50%;

        .finnhub-logo {
          height: 80px;
          align-self: center;
          .finnhub-logo-img {
            height: 100%;
          }
        }
      }
    }
  }
}

.finnhub-text {
  margin: 8px 0px 0px 0px;
  font-size: 12px;
  letter-spacing: 0.5px;
  font-weight: 300;
  position: relative;
  right: 90px;
}
