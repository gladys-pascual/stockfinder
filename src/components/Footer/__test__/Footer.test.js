import { render } from "@testing-library/react";
import Footer from "../Footer";

test("render Footer as expected", () => {
  const { container } = render(<Footer />);
  expect(container).toMatchSnapshot();
});
