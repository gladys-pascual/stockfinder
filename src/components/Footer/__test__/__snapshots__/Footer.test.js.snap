// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Footer as expected 1`] = `
<div>
  <footer
    class="footer-section"
  >
    <a
      class="finnhub"
      href="https://finnhub.io/docs/api"
      rel="noopener noreferrer"
      target="_blank"
    >
      <div
        class="finnhub-logo"
        href="https://finnhub.io/docs/api"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Finnhub logo"
          class="finnhub-logo-img"
          src="https://static.finnhub.io/img/finnhub_2020-05-09_20_51/logo/logo-gradient-thumbnail-trans.png"
        />
      </div>
      <p
        class="finnhub-text"
      >
        This website uses the Finnhub API.
      </p>
    </a>
    <ul
      class="footer-socials"
    >
      <li>
        <a
          href="https://github.com/gladys-pascual/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <i
            class="fab fa-github"
          >
            <span
              class="sr-only"
            >
              GitHub icon
            </span>
          </i>
        </a>
      </li>
      <li>
        <a
          href="https://dev.to/gladyspascual"
          rel="noopener noreferrer"
          target="_blank"
        >
          <i
            class="fab fa-dev"
          >
            <span
              class="sr-only"
            >
              Dev icon
            </span>
          </i>
        </a>
      </li>
      <li>
        <a
          href="https://www.linkedin.com/in/gladyspascual/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <i
            class="fab fa-linkedin"
          >
            <span
              class="sr-only"
            >
              LinkedIn icon
            </span>
          </i>
        </a>
      </li>
    </ul>
  </footer>
</div>
`;
